pkg/kubelet/kubelet_pods.go

```diff
@@ -675,6 +676,38 @@ func (kl *Kubelet) podIsTerminated(pod *v1.Pod) bool {
 	return false
 }
 
+func (kl *Kubelet) safeToDeletePod(pod *v1.Pod) bool {
```

Comment:
add some godoc here explaining what and why this exists as it will not be obvious to future folks... i know for a fact the relationship around the whole delete pod flow in kubelet is black magic to many.